streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignorTest.java

```diff
@@ -0,0 +1,264 @@
+
+    private final Node node0 = new Node(0, "node0", 1, "rack1");
+    private final Node node1 = new Node(1, "node1", 1, "rack2");
+    private final Node node2 = new Node(2, "node2", 1, "rack3");
+    private final Node[] replicas = new Node[] {node0, node1, node2};
+
+    private final PartitionInfo partitionInfo00 = new PartitionInfo(TOPIC0, 0, node0, replicas, replicas);
+    private final PartitionInfo partitionInfo01 = new PartitionInfo(TOPIC0, 1, node0, replicas, replicas);
+
+    private final TopicPartition tp00 = new TopicPartition(TOPIC0, 0);
+    private final TopicPartition tp01 = new TopicPartition(TOPIC0, 1);
+    private final TopicPartition tp10 = new TopicPartition(TOPIC1, 0);
+
+    private final UUID process0UUID  = UUID.randomUUID();
+    private final UUID process1UUID = UUID.randomUUID();
+
+    private final Subtopology subtopology1 = new Subtopology(1, "topology1");
+
+    private final MockTime time = new MockTime();
+    private final StreamsConfig streamsConfig = new StreamsConfig(configProps());
+    private final MockClientSupplier mockClientSupplier = new MockClientSupplier();
+
+    private final MockInternalTopicManager mockInternalTopicManager = new MockInternalTopicManager(
+            time,
+            streamsConfig,
+            mockClientSupplier.restoreConsumer,
+            false
+    );
+
+    private Map<String, Object> configProps() {
+        final Map<String, Object> configurationMap = new HashMap<>();
+        configurationMap.put(StreamsConfig.APPLICATION_ID_CONFIG, APPLICATION_ID);
+        configurationMap.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, USER_END_POINT);
+
+        final ReferenceContainer referenceContainer = new ReferenceContainer();
+        /*
+        referenceContainer.mainConsumer = consumer;
+        referenceContainer.adminClient = adminClient;
+        referenceContainer.taskManager = taskManager;
+        referenceContainer.streamsMetadataState = streamsMetadataState;
+        referenceContainer.time = time;
+        */
+        configurationMap.put(InternalConfig.REFERENCE_CONTAINER_PARTITION_ASSIGNOR, referenceContainer);
+        return configurationMap;
+    }
+
+    @Test
+    public void disableActiveSinceMissingClusterInfo() {
+        final Cluster metadata = new Cluster(
+            "cluster",
+            new HashSet<>(Arrays.asList(node0, node1, node2)),
+            new HashSet<>(Arrays.asList(partitionInfo00, partitionInfo01)),
+            Collections.emptySet(),
+            Collections.emptySet()
+        );
+
+        final Map<UUID, Map<String, Optional<String>>> processRacks = new HashMap<>();
+
+        processRacks.put(process0UUID , Collections.singletonMap("consumer1", Optional.of("rack1")));
+
+        final RackAwareTaskAssignor assignor = new RackAwareTaskAssignor(
+            metadata,
+            Collections.singletonMap(new TaskId(1, 1), new HashSet<>(Arrays.asList(tp00, tp10))),
```

Comment:
Should we read from all three partitions? Or only from `tp10`?

Reply:
For simplicity, this is just 1 task.